x-pack/plugins/task_manager/server/monitoring/task_run_statistics.ts

```diff
@@ -36,6 +36,16 @@ import { HealthStatus } from './monitoring_stats_stream';
 import { TaskPollingLifecycle } from '../polling_lifecycle';
 import { TaskExecutionFailureThreshold, TaskManagerConfig } from '../config';
 
+export enum TaskPersistence {
+  Recurring = 'recurring',
+  NonRecurring = 'non_recurring',
+  Ephemeral = 'ephemeral',
+}
+
+function persistenceOf(task: ConcreteTaskInstance) {
+  return task.schedule ? TaskPersistence.Recurring : TaskPersistence.NonRecurring;
+}
```

Comment:
As you can see here - we only ever identify a task as having a "recurring" or a "non-recurring" persistence.
"Ephemeral" tasks never get detected because they don't yet exist- this is plumbing in preparation for https://github.com/elastic/kibana/pull/97311.

If that PR isn't merged in 7.14, then we'll have to merge a small fix here removing ephemeral tasks from the output of this API.